src/common/connection.ts

```diff
@@ -36,29 +36,67 @@ export interface ConnectionOptions {
  */
 export type ConnectionUserOptions = Partial<ConnectionOptions>
 
+/**
+ * Ledger is used to store and reference ledger information that has been
+ * captured by the Connection class.
+ */
+class Ledger {
```

Comment:
This class names feels like a misnomer to me.  If this is about a connection, call it a connection? Is it a client of a ledger? A connection to a ledger? 

Calling it a `Ledger` makes it sound like a model object, and though I am having trouble reasoning about what this actually is, it's not a model? `LedgerMetadata` perhaps?

Reply:
perhaps `LedgerHistory`

Reply:
👍 will go with `LedgerHistory`

Reply:
Sounds good. Thank you for humoring my pedantic nature :)